pennylane_lightning/lightning_qubit/lightning_qubit2.py

```diff
@@ -34,24 +34,48 @@
 from pennylane.transforms.core import TransformProgram
 from pennylane.typing import Result, ResultBatch
 
+from ._state_vector import LightningStateVector
+from ._measurements import LightningMeasurements
 
-def dummy_simulate(
-    circuit,
-    rng=None,
-    c_dtype=np.complex128,
-    batch_obs=False,
-    mcmc=False,
-    kernel_name="Local",
-    num_burnin=100,
-):
-    return tuple(0.0 for _ in circuit.measurements)
 
+try:
+    # pylint: disable=import-error, no-name-in-module
+    from pennylane_lightning.lightning_qubit_ops import (
+        StateVectorC64,
+        StateVectorC128,
+    )
 
-def dummy_jacobian(circuit):
+    LQ_CPP_BINARY_AVAILABLE = True
+except ImportError:
+    LQ_CPP_BINARY_AVAILABLE = False
+
+
+def simulate(circuit: QuantumScript, dtype=np.complex128, debugger=None) -> Result:
+    """Simulate a single quantum script.
+
+    Args:
+        circuit (QuantumTape): The single circuit to simulate
+        dtype: Datatypes for state-vector representation. Must be one of
+            ``np.complex64`` or ``np.complex128``.
+        debugger (_Debugger): The debugger to use
+
+    Returns:
+        tuple(TensorLike): The results of the simulation
+
+    Note that this function can return measurements for non-commuting observables simultaneously.
+
+    """
+    state, is_state_batched = LightningStateVector(
```

Comment:
Is it in the plan to support native broadcasted parameters in lightning? If not, we can just assume the circuit does not have any broadcasted parameters and that the state is never broadcasted.

Reply:
Not in the foreseeable future, I will remove it.